src/lib/appConfig.ts

```diff
@@ -67,6 +67,7 @@ export const appConfig = {
   async read(): Promise<Config> {
     const result = await fs.readTextFile(await this.configFullPath())
     const config: Config = JSON.parse(result)
+    config.plot.sizeGB = toFixed(config.plot.sizeGB, 2)
```

Comment:
Not sure what this does, but it still seems wrong too.

```typescript
parseFloat(Number(num).toFixed(precision))
```

that you have in `toFixed` can still lead to numbers with a lot of digits after `.`, there is a reason why `Number.toFixed()` returns a string and not another float.

`toFixed` should return a string and be used directly in the layout template, not parsed back to float and reassigned into the config.

Reply:
how come toFixed can still lead to a number with lot of digits after `.` ? We have given a precision of 2, not something huge. I think the language can handle 2 digits for the precision and not return an unexpected case?

However, the changed line should have been `toFixed(Number(config.plot.sizeGB), 2)` fixing now.

Reply:
We are kind of in a hurry to get the release of 0.5.2, so I say, let's merge this change for now. And keep the discussion alive. Because refactoring it will take time, `toFixed` is used in many places, and we need to do more testing in other places if we want to change the signature of the function). 
If we decide on refactoring it, we can always do so in the future. What do you say @nazar-pc ?

Reply:
Floats are complicated in computing, for instance:
```
0.1+0.7
0.7999999999999999
```

So it would be preferrable to store integers whenever possible. Ideally in the config too and only converting to floats for UI purposes later.

Why does config store gigabytes as a string though? It seems like an odd design decision and the root of the problem.

Reply:
Leo designed it back then...